tor_flair/main.py

```diff
@@ -0,0 +1,114 @@
+    'our volunteers on your subreddit.\n\n'
+    'Simply set flair-text and / or flair-css in {} '
+    'and u/tor_flair_bot will get the changes the next time a volunteer accepts a '
+    'job for your subreddit.\n\n'
+    'If you have any questions or issues with the bot, please '
+    'send a modmail to r/TranscribersOfReddit and we\'ll get you sorted as quickly '
+    'as we can!'.format(settings_location)
+)
+
+
+settings_page = (
+    '; Set the flair text and css class below to whatever you want.  \n'
+    '; Both options are optional - whichever one is blank be left alone.  \n'
+    '; Example:  \n'
+    ';  \n'
+    '; flair-text: darn cool person  \n'
+    '; flair-css: coolperson  \n'
+    '\n\n'
+    'flair-text:\n\n'
+    'flair-css:'
+)
+
+
+def setup_wiki(subreddit):
+    subreddit.wiki.create(settings_location, content=settings_page)
+    subreddit.wiki.create(instructions_location, content=instructions_page)
+
+
+def get_config(subreddit):
+    return get_wiki_page(settings_location, config, subreddit=subreddit)
+
+
+def check_inbox(config):
+    logging.debug('Checking inbox...')
+    for item in config.r.inbox.unread():
+        if 'invitation to moderate' in item.subject.lower():
+            logging.info(
+                'Received offer to moderate {}!'.format(item.subreddit.display_name)
+            )
+            item.subreddit.mod.accept_invite()
+            setup_wiki(item.subreddit)
+            config.redis.sadd('flair_subreddits', item.subreddit.display_name)
+            item.mark_read()
+        if 'has been removed as a moderator from' in item.subject.lower():
+            logging.warning(
+                'I\'ve been kicked from {}!'.format(item.subreddit.display_name)
+            )
+            config.redis.srem('flair_subreddits', item.subreddit.display_name)
+            item.mark_read()
+
+
+def run(config):
+    check_inbox(config)
+    new_job = config.redis.lpop('flair_queue')
+    if not new_job:
+        time.sleep(10)
+        return
+    # format: user::subreddit
+    new_job = new_job.decode()  # bytes to str
+    user = config.r.redditor(
+        new_job[:new_job.index('::')]
+    )
+    subreddit = config.r.subreddit(new_job[new_job.index('::')+2:])
+
+    logging.info(
+        'Setting flair for {} in {}...'.format(user.name, subreddit.display_name)
+    )
+
+    subreddit_config = get_config(subreddit).split('\r\n')
+    subreddit_config = [y for y in subreddit_config if not y.startswith(';') and y != '']
+
+    text = subreddit_config[0][11:].strip()
+    css = subreddit_config[1][10:].strip()
+
+    if text == '':
+        # get the existing flair so we can write it back
+        # Also, PRAW, why is this a generator for one user?
+        text = [x for x in subreddit.flair(redditor=user)][0]['flair_text']
+    if css == '':
+        css = [x for x in subreddit.flair(redditor=user)][0]['flair_css_class']
+
+    subreddit.flair.set(user, text=text, css_class=css)
```

Comment:
This is probably better for the u/ToR part, but we should make sure the user doesn't have a flair yet since some people might have other custom flairs they prefer over the transcriber flair.

Reply:
This is handled by the section above. Whenever we set flair, both css class and text get set, so if none is specified in the settings we use `subreddit.flair()` to get the current set flair and write it along with our changes. For example:

wiki config file:
`flair-text: `
`flair-css: asdf`

current subreddit settings:
`flair_text: Cool Person`
`flair_css: coolperson`

because text is missing from the settings, we pull the existing flair text and so `subreddit.flair.set()` ends up with

`flair_text: Cool Person`
`flair_css: asdf`

Reply:
Right, but now it does it the other way around then? You take the config flair text/css first, and if those are not present you fill in the user's current flair.

Reply:
Yup!